test/unit/client/system/browserify.test.js

```diff
@@ -0,0 +1,303 @@
+'use strict';
+
+var path    = require('path'),
+  vm      = require('vm'),
+  fs      = require('fs'),
+  should  = require('should'),
+  ss      = require( '../../../../lib/socketstream');
+
+/**
+ * directories are defined with a trailing /
+ * JS and Coffee files are defined without
+ * Code should not be registered with their extension
+ *
+ *
+ * resolve:
+ require(X) from module at path Y
+ 1. If X is a core module,
+ a. return the core module
+ b. STOP
+ 2. If X begins with './' or '/' or '../'
+ a. LOAD_AS_FILE(Y + X)
+ b. LOAD_AS_DIRECTORY(Y + X)
+ 3. not relevant: LOAD_NODE_MODULES(X, dirname(Y))
+ 4. THROW "not found"
+
+ LOAD_AS_FILE(X)
+ 1. If X is a file, load X as JavaScript text.  STOP
+ 2. If X.js is a file, load X.js as JavaScript text.  STOP
+ 3. If X.json is a file, parse X.json to a JavaScript Object.  STOP
+ 4. If X.node is a file, load X.node as binary addon.  STOP
+
+ LOAD_AS_DIRECTORY(X)
+ 1. If X/package.json is a file,
+ a. Parse X/package.json, and look for "main" field.
+ b. let M = X + (json main field)
+ c. LOAD_AS_FILE(M)
+ 2. If X/index.js is a file, load X/index.js as JavaScript text.  STOP
+ 3. If X/index.json is a file, parse X/index.json to a JavaScript object. STOP
+ 4. If X/index.node is a file, load X/index.node as binary addon.  STOP
+ */
+describe('browserify', function() {
+
+  var browserify = fs.readFileSync( path.join(__dirname,'../../../..', 'lib/client/system/libs/browserify.js')),
+      browser;
+
+  describe('require()', function () {
+
+    beforeEach(function() {
+      browser = (function(){
+        var window = {document:{}};
+        window.window = window;
+        vm.runInNewContext(browserify, window);
+
+        return window;
+      })();
+    });
+
+    it('should be sane',function() {
+      should(browser.require).be.type('function');
+    });
+
+    it('should have a functioning require.modules.path.resolve',function() {
+      var bPath = browser.require.modules.path();
+
+      bPath.resolve('/').should.be.equal('/');
+      bPath.resolve('abc/def','./ghi').should.be.equal('abc/def/ghi');
+      bPath.resolve('foo/bar', '/tmp/file/', '..', 'a/../subfile').should.be.equal('/tmp/subfile');
+    });
+
+    it('should have functioning require.modules.path.dirname',function() {
+      var bPath = browser.require.modules.path();
+
+      bPath.dirname('/a/b/c/d').should.be.equal('/a/b/c');
+      //bPath.dirname('/a/b/c/d/').should.be.equal('/a/b/c/d');
+    });
+
+    it('should resolve /index directly or indirectly', function() {
+      browser.require.define('/a/b/c/d',
+        function(require, module, exports, __dirname, __filename) {
```

Comment:
'__filename' is defined but never used.<br>'__dirname' is defined but never used.<br>'exports' is defined but never used.<br>'module' is defined but never used.<br>'require' is defined but never used.